### ISSUE TITLE
`options` getting called on a `skip`

### ISSUE THREAD

**Issue by Panoplos** [apollographql/react-apollo#1280]
**Intended outcome:**

For illustration, use this HOC:

```es6
const withDependency = graphql(DependencyQuery,
  {
    skip: ({dependency}) => !dependency || !dependency.id,
    options: ({dependency}) => ({
      variables: {
        id: dependency.id
      }
    })
  }
)
```
I expect options to not be evaluated as a result of the `skip` test failing. 

**Actual outcome:**
`options` is called when `dependency` doesn't exist, resulting in a javascript prop access error.

**Version**
- apollo-client@1.4.16
- react-apollo@1.9.3

**Comment by simontegg** [comment]
This is a duplicate of #903 

There's a fix in v.2 but I'm unable to upgrade as a I depend on redux

**Comment by glasser** [comment]
This was fixed in #1181 but that didn't make it to 2.0 somehow. #1330 should resolve that.

**Comment by Panoplos** [comment]
Any update on this? It is still broken in `2.0.1`.

**Comment by Panoplos** [comment]
Nevermind...It looks like it is fixed in 2.0.1. (My vendor dll was not getting rebuilt.)
